rpcs3/Emu/Cell/lv2/sys_config.cpp

```diff
@@ -0,0 +1,436 @@
+﻿#include "stdafx.h"
+#include "Emu/System.h"
+#include "Emu/Memory/vm.h"
+#include "Emu/IdManager.h"
+
+#include "Emu/Cell/lv2/sys_event.h"
+#include "Emu/Cell/ErrorCodes.h"
+
+#include "sys_config.h"
+
+logs::channel sys_config("sys_config");
+
+
+// Enums
+template<>
+void fmt_class_string<sys_config_service_id>::format(std::string& out, u64 id)
+{
+	switch (static_cast<s64>(id))
+	{
+	case SYS_CONFIG_SERVICE_PADMANAGER   : out += "SYS_CONFIG_SERVICE_PADMANAGER"; return;
+	case SYS_CONFIG_SERVICE_PADMANAGER2  : out += "SYS_CONFIG_SERVICE_PADMANAGER2"; return;
+	case SYS_CONFIG_SERVICE_USER_LIBPAD  : out += "SYS_CONFIG_SERVICE_USER_LIBPAD"; return;
+	case SYS_CONFIG_SERVICE_USER_LIBKB   : out += "SYS_CONFIG_SERVICE_USER_LIBKB"; return;
+	case SYS_CONFIG_SERVICE_USER_LIBMOUSE: out += "SYS_CONFIG_SERVICE_USER_LIBMOUSE"; return;
+	}
+
+	if (id < 0)
+	{
+		id = (id & 0x7FFF'FFFF'FFFF'FFFFllu);
+		fmt::append(out, "SYS_CONFIG_SERVICE_USER_%llx", id);
```

Comment:
Instead of the last 2 lines consider:
```suggestion
		fmt::append(out, "SYS_CONFIG_SERVICE_USER_%llx", id & ~(1ull << 63));
```
Also, this is unreachable because u64 cannot be less than 0 (typo s64?).

Reply:
Done.